pkg/resources/management.cattle.io/v3/globalrolebinding/validator.go

```diff
@@ -220,3 +223,21 @@ func (a *admitter) validateGlobalRole(globalRole *v3.GlobalRole, fieldPath *fiel
 	}
 	return nil
 }
+
+func (a *admitter) validateNamespacedRules(request *admission.Request, rules map[string][]v1.PolicyRule) []error {
```

Comment:
Nit: I would recommend changing the name of this to indicate the escalation checks. You don't really need to validate the rules, more that your binding isn't giving excessive rules.